### PR TITLE
Contactus

### PR DESCRIPTION
**Pull request by Himanshu-tester**
# Overview
Added Test Scripts

# Technical Details
Added the contact page test script and the product details test script

# Reviewers
@abhilashcae

### CODE REVIEW

File: data/data.json [hunk 1]
```diff
@@ -8,5 +8,11 @@
   "postcode": "22030",
   "phone_number": "1234567890",
   "alias_text": "My address",
-  "new_email": "himans001234567000@gmail.com"
+  "new_email": "himans0012334@gmail.com",
```
Comment (abhilashcae) [inline data/data.json:11]: As mentioned earlier, let's make use of unique date each time.

File: data/data.json [hunk 2]
```diff
@@ -8,5 +8,11 @@
...
+  "contact_email": "himanshusethi002@gmail.com",
+  "upload_file": "C:\\Users\\Himanshu\\Desktop\\Himanshu_Resume.pdf",
```
Comment (abhilashcae) [inline data/data.json:13]: We shouldn't hardcode the path. Let's do this programmatically

File: logs/logfile.log [hunk 3]
```diff
@@ -1,40 +1,20 @@
...
-2020-08-27 20:27:22,564 :INFO :utilities.log :Clicked on the element submitAccount
-2020-08-27 20:27:22,566 :INFO :utilities.log :Entered Phone Number is 1234567890
-2020-08-27 20:27:19,068 :INFO :utilities.log :Drop down element id_country
-2020-08-27 20:27:19,119 :INFO :utilities.log :Entered City is Fairfax
-2020-08-27 20:27:19,120 :INFO :utilities.log :Entered Postal Code is 22030
-2020-08-27 20:27:19,351 :INFO :utilities.log :Entered text element phone_mobile
-2020-08-27 20:27:20,043 :INFO :utilities.log :Entered text element alias
-2020-08-27 20:27:22,564 :INFO :utilities.log :Clicked on the element submitAccount
-2020-08-27 20:27:22,566 :INFO :utilities.log :Entered Phone Number is 1234567890
+2020-09-07 16:20:23,428 :INFO :utilities.log :Clicked on the element //*[@class='logo img-responsive']
```
Comment (abhilashcae) [inline logs/logfile.log:1]: Again, this file should not be included in the PR. We don't need this to be checked in in github.

File: page_objects/item_details.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        super().__init__(driver)
+        self.driver = driver
+        self.data = DataRead.json_read('data.json')
+
+    search_field = LocatorStrategy.locator_by_id("search_query_top")
+    search_button = LocatorStrategy.locator_by_xpath("//button[@name='submit_search']")
+    more_options = LocatorStrategy.locator_by_xpath("//div[@class='product-container']/div[2]/div[2]/a[@title='View']")
+    select_size = LocatorStrategy.locator_by_id("group_1")
+    products = LocatorStrategy.locator_by_xpath("//div[@class='product-container']/div[1]/div/a/img")
+    prod_name = LocatorStrategy.locator_by_xpath("//*[@id='center_column']/ul/li[3]/div/div[2]/h5/a")
```
Comment (abhilashcae) [inline page_objects/item_details.py:19]: Avoid using //* unless absolutely required. It may skew up the xpaths.

Comment (abhilashcae) [inline page_objects/item_details.py:19]: Avoid using //* unless absolutely required. It may skew up the xpaths.

File: page_objects/item_search.py [hunk 5]
```diff
@@ -1,32 +1,32 @@
...
+        # d = DataRead()
         self.data = DataRead.json_read('data.json')
+        self.log = self.logger()
 
     search_field = LocatorStrategy.locator_by_id("search_query_top")
     search_button = LocatorStrategy.locator_by_xpath("//button[@name='submit_search']")
     product = LocatorStrategy.locator_by_xpath("//div[@class='product-container']/div[1]/div/a/img")
-    add_to_cart = LocatorStrategy.locator_by_xpath("//div[@class='product-container']/div[2]/div[2]/a[@title='Add to cart']")
+    add_to_cart = LocatorStrategy.locator_by_xpath(
+        "//div[@class='product-container']/div[2]/div[2]/a[@title='Add to cart']")
```
Comment (abhilashcae) [inline page_objects/item_search.py:21]: we definitely can write better xpaths at many places including this.
this add to cart xpath can be //p[@id="add_to_cart"]/button

File: tests/test_productdetails.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import pytest
```
Comment (abhilashcae) [inline tests/test_productdetails.py:1]: Since we're following snake case naming convention here, we should have underscores after each word. Ex: test_product_details.

Comment (Himanshu-tester) [reply tests/test_productdetails.py:1]: Updated with the snake case naming convention.

File: data/data.json (hunk 2 above)
Comment (Himanshu-tester) [reply data/data.json:13]: Updated with the file path in the config file.

File: data/data.json (hunk 1 above)
Comment (Himanshu-tester) [reply data/data.json:11]: Not sure what to use so I gave something like this.
